skimage/future/graph/rag.py

```diff
@@ -321,6 +382,100 @@ def rag_mean_color(image, labels, connectivity=2, mode='distance',
     return graph
 
 
+def rag_generic(image, labels, connectivity=2, describe_func=describe_node,
+                weight_func=mean_color, extra_arguments=[],
+                extra_keywords={'sigma':255.0, 'mode':'distance'}):
```

Comment:
Thanks for the feedback.  
1. Will rename to region_adjacency_graph.
2. Can reorder parameters.
3.  Can change default, have no preference.
4. OK to default callbacks to none. Was just trying to duplicate rag_mean_color.
5. `node_descriptor` is a better name.  Having a `edge_descriptor` makes sense.
6.  `weight_func` assumes that the nodes have been described.  Will think about encapsulating both into a single function.